From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct iwl_trans
Patch-mainline: never, kabi

In 3.12.34, commit 9546673d9dc0c6c642d7f4740622d15d7fb980f9 (iwlwifi:
configure the LTR), upstream commit
9180ac50716a097a407c6d7e7e4589754a922260 added ltr_enabled bool to
struct iwl_trans. It indeed changed the layout of that structure.

However the structure is iwl-internal only, so protect the change in
the member the usual way, by #ifdef __GENKSYMS__.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/net/wireless/iwlwifi/iwl-trans.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/net/wireless/iwlwifi/iwl-trans.h
+++ b/drivers/net/wireless/iwlwifi/iwl-trans.h
@@ -509,7 +509,9 @@ struct iwl_trans {
 	u8 rx_mpdu_cmd, rx_mpdu_cmd_hdr_size;
 
 	bool pm_support;
+#ifndef __GENKSYMS__
 	bool ltr_enabled;
+#endif
 
 	/* The following fields are internal only */
 	struct kmem_cache *dev_cmd_pool;
